ca/certificate-authority.go

```diff
@@ -253,10 +254,10 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	hostNames := make([]string, len(csr.DNSNames))
 	copy(hostNames, csr.DNSNames)
 	if len(csr.Subject.CommonName) > 0 {
-		commonName = csr.Subject.CommonName
-		hostNames = append(hostNames, csr.Subject.CommonName)
+		commonName = strings.ToLower(csr.Subject.CommonName)
```

Comment:
I would recommend moving this whole chunk (default-filling the CN field) below where we say `hostNames = core.UniqueNames(hostNames)`. That way we don't have to duplicate the ToLower logic. also, the UniqueNames call should be commented saying "this is also responsible for lowercasing the names."